lib/build-plugin.js

```diff
@@ -33,6 +33,12 @@ function buildPlugin(Client, pluginOpts) {
 
     const secrets = await pProps(opts.secrets, (value) => client.get(value), { concurrency })
 
+    if (Array.isArray(opts.secrets)) {
+      for (let i = 0; i < opts.secrets.length; i++) {
+        secrets[opts.secrets[i]] = secrets[i]
```

Comment:
Moving this logic before the `pProps` call to compose an input parameter, would remove the index properties from `fastify.secrets`.

I think it would be more compliant to the principle of least astonishment

Reply:
I have these index properties here for backwards compatibility. Do we not want to maintain backwards compatibility? This way, code like `fastify.secrets[0]` would still work